Code/EnginePlugins/OpenXRPlugin/OpenXRSingleton.h

```diff
@@ -0,0 +1,207 @@
+#pragma once
+
+#include <Core/Graphics/Camera.h>
+#include <Foundation/Configuration/Singleton.h>
+#include <GameEngine/XR/XRInterface.h>
+#include <GameEngine/XR/XRInputDevice.h>
+#include <OpenXRPlugin/Basics.h>
+#include <OpenXRPlugin/OpenXRIncludes.h>
+#include <RendererCore/Pipeline/Declarations.h>
+#include <RendererCore/Shader/ConstantBufferStorage.h>
+#include <RendererFoundation/Descriptors/Descriptors.h>
+#include <RendererFoundation/Resources/RenderTargetSetup.h>
+#include <RendererFoundation/Device/SwapChain.h>
+
+class ezOpenXRInputDevice;
+
+struct ezGameApplicationExecutionEvent;
+
+EZ_DEFINE_AS_POD_TYPE(XrSwapchainImageD3D11KHR);
+EZ_DEFINE_AS_POD_TYPE(XrViewConfigurationView);
+EZ_DEFINE_AS_POD_TYPE(XrEnvironmentBlendMode);
+
+//#TODO: Make use of this to abstract render api.
```

Comment:
This isn't used yet? @C-Core

Reply:
I guess this was the first try to get the native d3d texture into our render pipeline. I haven't looked at the OpenXR implementation in detail by this class is most likely not needed anymore.

---
In reply to: [423519047](https://github.com/ezEngine/ezEngine/pull/212#discussion_r423519047) [](ancestors = 423519047)

Reply:
Removing it.